### PR TITLE
[WIP] implement godbc backend

### PR DESCRIPTION
**Pull request by Habbie**
- [x] add charset to docs (see test-auth)
- [ ] add godbc_sqlite3 to start-test-stop help
- [ ] hide passwords during test run
- [ ] make it possible to log SUCCESS_WITH_INFO instead of erroring on it?
- [ ] via @cmouse: Perhaps not cast everything to long, but handle binds of various integer types correctly?
- [ ] fix `set-tsig-key-query` default for mssql
- [ ] don't silently truncate SELECT results over 128kbyte per cell
- [ ] make SELECT results null-safe
- [ ] have one final look at the SQL_NO_DATA/NOCOUNT/multiple result set stuff

### CODE REVIEW

File: modules/godbcbackend/sodbc.cc [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+    testResult( result, SQL_HANDLE_STMT, d_statement, "Binding parameter.");
+    d_paridx++;
+
+    return this;
+  }
+  SSqlStatement* bind(const string& name, uint32_t value) { return bind(name, (long)value); }
+  SSqlStatement* bind(const string& name, int value) { return bind(name, (long)value); }
+  SSqlStatement* bind(const string& name, unsigned long value) { return bind(name, (long)value);}
+  SSqlStatement* bind(const string& name, long long value) { return bind(name, (long)value); };
+  SSqlStatement* bind(const string& name, unsigned long long value) { return bind(name, (long)value); }
```
Comment (cmouse) [inline modules/godbcbackend/sodbc.cc]: Perhaps not cast everything to long, but handle them correctly? This might bite later on.

**Comment by Habbie** [comment]
We drop the info from SQL_SUCCESS_WITH_INFO - in the one situation we observed info we actually wanted to treat this as an error. Considering treating it as an error in general!

**Comment by Habbie** [comment]
Note: to be able to import `hightxt.test.com` with `pdnssec load-zone`, use `ServerName` instead of `Server` in odbc.ini, then add that server to `freetds.conf` (with `host`/`port`/`tds` version) so you can add `client charset = UTF-8`

File: modules/godbcbackend/sodbc.cc (hunk 1 above)
Comment (Habbie) [inline modules/godbcbackend/sodbc.cc]: Do you foresee any biting that will go unnoticed?

**Comment by Habbie** [comment]
TLDR please review.

I'd like to merge this ASAP, even with some of the boxes unchecked, can just file a ticket for that. Please report any actual merge breakers if you have them.

**Comment by Habbie** [comment]
(I will update the commit message before merge)

**Comment by zeha** [comment]
wrote a ton of nits; in general killing lots of commented out lines may help?

**Comment by Habbie** [comment]
all nits from @zeha fixed
